utopia-vscode-common/src/fs/fs-core.ts

```diff
@@ -10,29 +10,42 @@ let thisDBName: string
 
 const StoreExistsKey = '.store-exists'
 
+/* eslint-disable-next-line @typescript-eslint/no-empty-function */
+let resolveFirstInitialize: () => void = () => {}
+const firstInitialize = new Promise<void>((resolve) => {
+  resolveFirstInitialize = resolve
+})
```

Comment:
we have a util to create a promise that can be externally resolved, called `defer` in `utils.ts`.
it encapsulates the promise and the externally accessible resolver function, so it lets you avoid creating a `let` in the module scope and mutating it.

I believe the code to use it would be this:
```
const firstInitialize = defer<void>()
```

and instead of calling `resolveFirstInitialize()`, you'd write
```
firstInitialize.resolve()
```

Reply:
I _knew_ we were using this exact pattern somewhere else!